tab-bot-sso-with-backend/.vscode/tasks.json

```diff
@@ -209,36 +209,32 @@
             }
         },
         {
-            "id": "prepare-local-environment",
+            "id": "prepare-teams-app-dependencies",
             "type": "command",
-            "command": "fx-extension.debug-task-prepare-local-environment",
+            "command": "fx-extension.prepare-app-dependencies",
             "args": {
                 "enabledSteps": [
-                    "registerBot",
-                    "registerSSOAAD",
-                    "buildTeamsManifest",
+                    "registerBot", // Register Bot
+                    "registerSSOAAD", // Register AAD app for SSO
+                    "generateEnv", // Generate environment variables for local services
                 ],
-                "tab": {
-                    "port": 53000,
-                    "envSavePath": "${teamsfx:workspaceFolder}/tabs/.env.teamsfx.local"
-                },
-                "azureFunctions": {
-                    "port": 9229,
-                    "envSavePath": "${teamsfx:workspaceFolder}/api/.env.teamsfx.local"
-                },
-                "bot": {
-                    "port": 3978,
-                    "envSavePath": "${teamsfx:workspaceFolder}/bot/.env.teamsfx.local",
-                    "messagingEndpoint": "${teamsfx:botTunnelEndpoint}/api/messages"
+                "appConfigs": {
+                    "tabBaseUrl": "https://localhost:53000",
+                    "aadObjectId": "${teamsfx:aadObjectId}",
+                    "aadClientId": "${teamsfx:aadClientId}",
+                    "aadClientSecret": "${teamsfx:aadClientSecret}",
+                    "botId": "${teamsfx:botId}",
+                    "botPassword": "${teamsfx:botPassword}",
+                    "botMessagingEndpoint": "${teamsfx:botTunnelEndpoint}/api/messages",
```

Comment:
For placeholders and variable references.
- If the variable is provided to be referenced by user, e.g., `workspaceFolder`, `botTunnelEndpoint`, use as `teamsfx:xxx`.
- But, if the variable is only used internally, e.g., `botPassword`, just use default placeholder like `${auto}`. Since we do not expect user to reference such value.

Reply:
What's the meaning of "referenced by user"?

Reply:
User can use the variables in teamsfx tasks. E.g., `${teamsfx:workspaceFolder}` can be used to represent the root folder, `${teamsfx:botTunnelEndpoint}` can be used to customize bot routing. But, when will user use `${teamsfx:botPassword}`?

Reply:
Yeah, make sense. Updated.